kmod/core/core.c

```diff
@@ -314,27 +256,47 @@ static const struct stacktrace_ops kpatch_print_trace_ops = {
  */
 static int kpatch_verify_activeness_safety(struct kpatch_module *kpmod)
 {
+
+#define MAX_STACK_TRACE_DEPTH   64
+
 	struct task_struct *g, *t;
+        unsigned long *entries;
+        int i, j;
 	int ret = 0;
 
-	struct kpatch_backtrace_args args = {
-		.kpmod = kpmod,
-		.ret = 0
-	};
+        entries = kmalloc(MAX_STACK_TRACE_DEPTH * sizeof(*entries), GFP_KERNEL);
+        if (!entries)
+                return -ENOMEM;
 
 	/* Check the stacks of all tasks. */
 	do_each_thread(g, t) {
-		dump_trace(t, NULL, NULL, 0, &kpatch_backtrace_ops, &args);
-		if (args.ret) {
-			ret = args.ret;
-			pr_info("PID: %d Comm: %.20s\n", t->pid, t->comm);
-			dump_trace(t, NULL, (unsigned long *)t->thread.sp,
-				   0, &kpatch_print_trace_ops, NULL);
-			goto out;
+
+		struct stack_trace trace = {
+			.nr_entries	= 0,
+			.max_entries	= MAX_STACK_TRACE_DEPTH,
+			.entries	= entries,
+			.skip		= 0,
+		};
+
+                save_stack_trace_tsk(t, &trace);
```

Comment:
Will do.  BTW, I picked 64 since that was the value used in the proc stack interface.  It could just as easily be a module parameter (default to 64) if you think it should be more flexible.